CMakeLists.txt

```
@@ -975,6 +975,20 @@ function(add_cxx_flags_if_supported var)
   set(${var} "${${var}}" PARENT_SCOPE)
 endfunction()
 
+function(add_compile_warnings_flags lib)
+  if (USE_FAIL_ON_WARNINGS)
+    if (MSVC)
+      target_compile_options(arango PRIVATE /WX /D_WINSOCK_DEPRECATED_NO_WARNINGS)
```

Should here also be `lib` ?
```suggestion
      target_compile_options(${lib} PRIVATE /WX /D_WINSOCK_DEPRECATED_NO_WARNINGS)
```

---

Yes!